### PR TITLE
Swath aggregator CLI

### PR DESCRIPTION
**Pull request by claytharrison**
Adds a utility for aggregating collections of swaths by regular time intervals. CLI is accessible via `ascat_temporal_swath_agg`.

Also adds the swath file generator from #61.

### CODE REVIEW

File: src/ascat/aggregate/aggregators.py [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+        self.timedelta = pd.Timedelta(t_delta)
+        self.agg = agg
+        self.product = product
+        self.collection = rat.SwathFileCollection.from_product_id(
+            Path(filepath), product
+        )
+        self.grid = self.collection.grid
+        self.data = None
+        self.agg_vars = [
+            "surface_soil_moisture",
```
Comment (sebhahn) [inline src/ascat/aggregate/aggregators.py:76]: maybe only use surface_soil_moisture and backscatter40 for now

File: src/ascat/aggregate/aggregators.py [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+            )
+            chunk_end_str = (
+                np.datetime64(ds.attrs["end_time"])
+                .astype(datetime.datetime)
+                .strftime("%Y%m%d%H%M%S")
+            )
+            ds["location_id"] = ds["location_id"].astype(int)
+            ds = self._set_metadata(ds)
+            ds.to_netcdf(
+                f"{out_dir}/ascat_{self.agg}_{chunk_start_str}_{chunk_end_str}.nc",
```
Comment (sebhahn) [inline src/ascat/aggregate/aggregators.py:111]: maybe add spatial sampling and/or product id

File: src/ascat/aggregate/interface.py [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+        '-t_delta',
+        metavar='T_DELTA',
+        help='Time period for aggregation (e.g. 1D, 1W, 1M, 1Y, 2D, 3M, 4Y, etc.)'
+    )
+    parser.add_argument(
+        '-agg',
+        metavar='AGG',
+        help='Aggregation'
+    )
+    parser.add_argument(
```
Comment (sebhahn) [inline src/ascat/aggregate/interface.py:61]: this should come from the input filename

Comment (sebhahn) [reply src/ascat/aggregate/interface.py:61]: or metadata

**Review by sebhahn — CHANGES_REQUESTED** [review]
good job!

**Comment by claytharrison** [comment]
Now it grabs a single swath filename from the filepath, and checks it against the products that we have metadata for using a regex to get the appropriate product id. Possible product ids right now are `H129`, `H129_V1.0`, `H121_V1.0`, `H122`, `SIG0_6.25`, `SIG0_12.5`. To use this on other data, the info will have to be added to `read_native/xarray_io`: both a class that inherits from `SwathIOBase` for reading, as well as an entry to `swath_fname_regex_lookup` so that the ioclass will be findable from a single filename.

**Comment by claytharrison** [comment]
Everything here should be ready to merge, then I just need to update the changelog

**Comment by claytharrison** [comment]
Updated the changelog here - all is ready to merge.

**Review by sebhahn — APPROVED** [review]

**Pull request merged**